src/CachedFactory.test.ts

```diff
@@ -47,4 +47,18 @@ describe("CachedFactory", () => {
 
 		expect(getter.mock.calls).toEqual([["key-first"], ["key-second"]]);
 	});
+
+	it("clears the cache when .clear is executed", () => {
+		const getter = vi.fn().mockImplementation((key: string) => Symbol(key));
```

Comment:
The investigation around Symbols is really cool, but IMO there's a much more straightforward way to do this: [Vitest mocks](https://vitest.dev/guide/mocking.html).

```ts
const getter = vi.mockResolvedValueOnce("first").mockResolvedValueOnce("second");
```

The amount of code isn't significantly less, but I think `Symbol`s are an obscure enough topic for most potential contributors that it'd be better to avoid them.

Thoughts?

Reply:
Oh! I see it now.
Makes sense. I'll make the update.
Thanks.